Core/ComponentBaseClasses/elxTransformBase.hxx

```diff
@@ -388,16 +388,36 @@ TransformBase<TElastix>::ReadFromFile()
        * is not the same as this transform parameter file. Otherwise,
        * we will have an infinite loop.
        */
-      std::string fullFileName1 = itksys::SystemTools::CollapseFullPath(fileName);
-      std::string fullFileName2 = itksys::SystemTools::CollapseFullPath(configuration.GetParameterFileName());
-      if (fullFileName1 == fullFileName2)
+
+      const std::string configurationParameterFileName = configuration.GetParameterFileName();
+
+      if (itksys::SystemTools::CollapseFullPath(fileName) ==
+          itksys::SystemTools::CollapseFullPath(configurationParameterFileName))
       {
         itkExceptionMacro("ERROR: The InitialTransformParameterFileName is identical to the current "
                           "TransformParameters filename! An infinite loop is not allowed.");
       }
 
-      /** We can safely read the initial transform. */
-      this->ReadInitialTransformFromFile(fileName.c_str());
+      const auto lastConfigurationParameterFilePathSeparator = configurationParameterFileName.find_last_of("\\/");
```

Comment:
does itksys not have a robust function for what you are trying to do here?

Reply:
> does itksys not have a robust function for what you are trying to do here?

Good question 😃 I found some itksys functions that do similar things, by splitting a path to its components, and then joining them again: 
```
  static std::vector<std::string> SplitString(const std::string& s,
                                              char separator = '/',
                                              bool isPath = false);
  static std::string Join(const std::vector<std::string>& list,
                          const std::string& separator);
```
But as far as I can see, it involves extra manual programming, it doesn't get much easier than what it does now: Just find the last separator (slash or backslash) in the current transform file path (by `std::string::find_last_of`), and then concatenate the current path _until that last separator_ with the string specified by "InitialTransformParameterFileName".

Reply:
ok, thanks! I [just](https://github.com/SuperElastix/elastix/compare/0e96c2f898cc38e3b2e2ed784d6caec446862c6b..088ec1100244df68c313cad23848999764b69ec2) added some more comment, and also made the code clearer by introducing a local variable, `isAbsoluteFilePath`. Hope it's clear enough now!